### PR TITLE
Login

### PR DESCRIPTION
**Pull request by jfeng5529**
## Description

Basic login flow complete

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
  - [ ] I wrote a design doc: *Replace this with a link to your design doc's short link*
  - [ ] I got input from a developer, specifically from: *Replace with the names of who gave advice*
  - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/rishabnayak/project-engage/issues
[Contributor Guide]: https://github.com/rishabnayak/project-engage/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,8 @@
   "dependencies": {
     "core-js": "^3.6.5",
     "firebase": "^8.2.10",
+    "googleapis": "^39.2.0",
+    "install": "^0.13.0",
```
Comment (rishabnayak) [inline package.json:14]: remove unnecessary dependencies

File: functions/package.json [hunk 2]
```diff
@@ -13,9 +13,12 @@
   },
   "main": "index.js",
   "dependencies": {
+    "@google-cloud/local-auth": "^1.0.1",
```
Comment (rishabnayak) [inline functions/package.json:16]: unnecessary

File: functions/package.json [hunk 3]
```diff
@@ -13,9 +13,12 @@
...
     "firebase-admin": "^9.2.0",
     "firebase-functions": "^3.11.0",
     "glob": "^7.1.6",
+    "googleapis": "^68.0.0",
+    "nodemailer": "^6.5.0",
```
Comment (rishabnayak) [inline functions/package.json:21]: this is unused?

**Review by rishabnayak — CHANGES_REQUESTED** [review]
please fix before merge!

File: functions/package.json (hunk 3 above)
Comment (jfeng5529) [reply functions/package.json]: googleapis is needed for the email sending but I removed the nodemailer

**Review by rishabnayak — APPROVED** [review]

**Pull request merged**